mm/kfence/report.c

```diff
@@ -132,6 +132,15 @@ void kfence_report_error(unsigned long address, const struct kfence_metadata *me
 	int num_stack_entries = stack_trace_save(stack_entries, KFENCE_STACK_DEPTH, 1);
 	int skipnr = get_stack_skipnr(stack_entries, num_stack_entries, type);
 
+	lockdep_assert_held(&meta->lock);
+	/*
+	 * Disable lockdep, as it might report on potential deadlock depending
+	 * on where printk() is called; this is unfortunately expected and
```

Comment:
The "expected and unavoidable" part is still lacking some information. Is this report expected because it's a real deadlock? If it is unavoidable, could it break us?

Reply:
Yes, it could break us, if a report is generated from the wrong place in the kernel. But this problem exists in all debugging tools (in fact when I had kmemleak enabled, I got lockdep warnings about kmemleak when it started printing things).

Calling printk from certain parts of the kernel may lead to deadlock, e.g. when called from the scheduler. I think that's what lockdep has been warning about. I had the same problem in KCSAN, and all we can do is (other than disable the tool in those parts of the kernel), is disable lockdep and ensure that those bits of the kernel don't have real bugs.

It's a known problem: https://lore.kernel.org/linuxppc-dev/20200623202404.GE2483@worktop.programming.kicks-ass.net/

I'll try to add comment that explains it better.